### PR TITLE
Docker rules engine issue 116

### PR DESCRIPTION
**Pull request by ethanstrominger**
See issue #116 

Docker will only process sub directories so it needed to be above rules-engine and heat-stack directory.  

I created a copy of the original Dockerfile to /Dockerfile-original to make comparison easier.

### CODE REVIEW

**Comment by thadk** [comment]
Hi @ethanstrominger thanks so much for this!

I just want to note that though we didn't get to check the final insertion-into-docker step, hypothetically in this PR from 10 days ago https://github.com/codeforboston/home-energy-analysis-tool/pull/176 gets the rules-engine built python egg into the GitHub Action (GHA) artifacts. Then the vitest uses it from the GHA artifacts. I figured we could get that into docker as it was building. Which approach or mix of approaches do you think we should ultimately use? Either way the versions should match which was our worry with our first other idea.

Sorry I didn't get back to update that original issue with the progress 🫤! 

We can totally merge in the docker way: my only tiny reservation is on extra divergence from [our docker deploy template](https://github.com/epicweb-dev/epic-stack/blob/main/.github/workflows/deploy.yml) which may yet get new features upstream. 
There is not any automated upgrade process though so just a bit of extra hand merging later. Also we might need to double check the .dockerignore changes.

**Comment by ethanstrominger** [comment]
Thad - Whatever you think is best.  I don't know enough about the other way to judge. With the docker approach I can test without checking into github.

**Comment by thadk** [comment]
That's a great point about being able to test locally! Let's use this.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+COPY --from=deps /myapp/heat-stack/node_modules /myapp/heat-stack/node_modules
+
+ADD heat-stack/prisma .
+RUN npx prisma generate
+
+ADD heat-stack/. .
+RUN npm run build
+
+FROM python:3.12.3-slim-bookworm as rules
```
Comment (alanisaac) [inline Dockerfile:40]: `minor`: I wonder if this should be named `rules-engine-build` or similar and the existing `build` above should be `heat-stack-build` or similar?  Just to clarify that _both_ of these containers are for the build process and not the final image?

**Review by thadk — CHANGES_REQUESTED** [review]
Please test on `dev` branch and get deploy stage working to heat-stack-staging.fly.dev .

Particularly you'll prolly have to adjust these paths: https://github.com/codeforboston/home-energy-analysis-tool/blob/76c568726726063e4f69354538b8b216a9ff239a/heat-stack/.github/workflows/deploy.yml#L160-L163

File: Dockerfile (hunk 1 above)
Comment (ethanstrominger) [reply Dockerfile]: I made the changes.  Yes, this is an intermediate build process.  heat-stack.yaml in github actions creates the final image.

**Pull request closed without merging**